test/unit/SmartContracts/proposers.unit.test.mjs

```diff
@@ -0,0 +1,427 @@
+
+    await ProposersInstance.deRegisterProposer();
+
+    expect((await state.getProposer(addr1.address)).thisAddress).to.equal(
+      ethers.constants.AddressZero,
+    );
+  });
+
+  it('should not deregister a non existing proposer', async function () {
+    expect((await state.getProposer(addr1.address)).thisAddress).to.equal(
+      ethers.constants.AddressZero,
+    );
+
+    await expect(ProposersInstance.deRegisterProposer()).to.be.revertedWith(
+      'Proposers: Not a proposer',
+    );
+  });
+
+  it('should not withdrawStake as a proposer', async function () {
+    expect((await state.getProposer(addr1.address)).thisAddress).to.equal(
+      ethers.constants.AddressZero,
+    );
+    const registerProposerTx = await ProposersInstance.registerProposer('url', 100, { value: 100 });
+    const receiptRegister = await registerProposerTx.wait();
+
+    const eventTransfer = receiptRegister.events.find(
+      event => event.event === 'NewCurrentProposer',
+    );
+    const [proposer] = eventTransfer.args;
+
+    expect(proposer).to.equal(addr1.address);
+
+    expect((await state.getProposer(addr1.address)).thisAddress).to.equal(addr1.address);
+
+    await expect(ProposersInstance.withdrawStake()).to.be.revertedWith(
+      'Proposers: Cannot withdraw while staking as proposer',
+    );
+  });
+
+  it('should not withdrawStake: Too soon to withdraw the stake', async function () {
+    expect((await state.getProposer(addr1.address)).thisAddress).to.equal(
+      ethers.constants.AddressZero,
+    );
+    const registerProposerTx = await ProposersInstance.registerProposer('url', 100, { value: 100 });
+    const receiptRegister = await registerProposerTx.wait();
+
+    const eventTransfer = receiptRegister.events.find(
+      event => event.event === 'NewCurrentProposer',
+    );
+    const [proposer] = eventTransfer.args;
+
+    expect(proposer).to.equal(addr1.address);
+
+    expect((await state.getProposer(addr1.address)).thisAddress).to.equal(addr1.address);
+
+    await ProposersInstance.deRegisterProposer();
+
+    await expect(ProposersInstance.withdrawStake()).to.be.revertedWith(
+      'Proposers: Too soon to withdraw the stake',
+    );
+  });
+
+  it('should withdrawStake', async function () {
+    expect((await state.getProposer(addr1.address)).thisAddress).to.equal(
+      ethers.constants.AddressZero,
+    );
+    const registerProposerTx = await ProposersInstance.registerProposer('url', 100, { value: 100 });
+    const receiptRegister = await registerProposerTx.wait();
+
+    const eventTransfer = receiptRegister.events.find(
+      event => event.event === 'NewCurrentProposer',
+    );
+    const [proposer] = eventTransfer.args;
+
+    expect(proposer).to.equal(addr1.address);
+
+    expect((await state.getProposer(addr1.address)).thisAddress).to.equal(addr1.address);
+
+    await ProposersInstance.deRegisterProposer();
+
+    await ethers.provider.send('evm_increaseTime', [604800]); // + 1 week
```

Comment:
This only works for ganache. We should write tests that work in all environments. So then we should wait for the blocks after challenge period or skip this test if it's not ganache.

Reply:
fixed, I added the check on chain id, and the test is run only in case of ganache or hardhat, otherwise the test is skipped